### PR TITLE
Flesh out GCP documentation for installation and project definition

### PR DESCRIPTION
**Pull request by lathanh**

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -193,5 +194,35 @@ especially over a slower internet connection as it is downloading and building a
...
+   user can install the `gcloud CLI <https://cloud.google.com/sdk/docs/install>`_. The following
+   documentation will assume use of a Service Account.
+
+Per-developer setup
+...................
+One-time setup that each developer needs to do on the workstation from which they'll launch and
+manage BuildStockBatch runs.
+
+1. Install Docker. This is needed by the script to manage Docker images (pull, push, etc).
+2. Install OpenStudio using the :ref:`local-install` instructions above.
```
Comment (nweires) [inline docs/installation.rst]: OpenStudio isn't required on the local machine.

File: docs/installation.rst [hunk 2]
```diff
@@ -193,5 +194,35 @@ especially over a slower internet connection as it is downloading and building a
...
-TODO: Add any GCP-specific installation instructions
-TODO: Need to install and run docker?
+Shared, one-time GCP setup
+..........................
+One-time GCP setup shared by all users.
+
+1. If needed, create a GCP Project. The following steps will occur in that project.
+2. Create a registry in Artifact Registry (to store Docker images).
+3. Create a GCS Bucket (that will store simulation and postprocessing output). Alternatively, each
+   user can create and use their own bucket.
```
Comment (nweires) [inline docs/installation.rst]:
```suggestion
2. [Create a repository](https://cloud.google.com/artifact-registry/docs/repositories/create-repos) in Artifact Registry (to store Docker images).
3. [Create a Google Cloud Storage Bucket](https://cloud.google.com/storage/docs/creating-buckets) (that will store simulation and postprocessing output). Alternatively, each
   user can create and use their own bucket.
```

I'd like to replace these steps with a single "apply the terraform config" step. But for now, links to the relevant docs are helpful.

File: docs/installation.rst [hunk 3]
```diff
@@ -193,5 +194,35 @@ especially over a slower internet connection as it is downloading and building a
...
+4. Create a Service Account. Alternatively, each user can create their own service account, or each
```
Comment (nweires) [inline docs/installation.rst:205]: We may want to include a list of the minimum set of permissions this service account needs. (Or a TODO to add that when we know the full list.)

File: docs/installation.rst [hunk 4]
```diff
@@ -193,5 +194,35 @@ especially over a slower internet connection as it is downloading and building a
...
+   user can install the `gcloud CLI <https://cloud.google.com/sdk/docs/install>`_. The following
+   documentation will assume use of a Service Account.
+
+Per-developer setup
+...................
+One-time setup that each developer needs to do on the workstation from which they'll launch and
+manage BuildStockBatch runs.
+
+1. Install Docker. This is needed by the script to manage Docker images (pull, push, etc).
```
Comment (nweires) [inline docs/installation.rst]:
```suggestion
1. [Install Docker](https://www.docker.com/get-started/). This is needed by the script to manage Docker images (pull, push, etc).
```

File: docs/project_defn.rst [hunk 5]
```diff
@@ -222,8 +222,39 @@ on the `AWS Batch <https://aws.amazon.com/batch/>`_ service.
...
+    *  ``bucket``: The Cloud Storage bucket this project will use for simulation output and
+       processed data storage.
+    *  ``prefix``: The Cloud Storage prefix at which the data will be stored.
+
+*  ``region``: The GCP region in which the batch will be run and of the Artifact Registry.
+*  ``use_spot``: true or false. Defaults to false if missing. This tells the project
+   to use `Spot VMs <https://cloud.google.com/spot-vms>`_ for data
+   simulations, which can reduce costs by up to 91%.
+*  ``batch_array_size``: Number of concurrent simulations to run. Max: 10000.
+*  ``notifications_email``: Email to notify you of simulation completion.
```
Comment (nweires) [inline docs/project_defn.rst]: This doesn't work yet, so let's leave a TODO to update the email details when it does. I'm not sure that the part about having to accept emails will be the same as on AWS.

File: docs/installation.rst (hunk 1 above)
Comment (lathanh) [reply docs/installation.rst]: Done.

File: docs/installation.rst (hunk 2 above)
Comment (lathanh) [reply docs/installation.rst]: Done (but with RST syntax)

File: docs/installation.rst (hunk 4 above)
Comment (lathanh) [reply docs/installation.rst]: Done (but with RST syntax).

File: docs/project_defn.rst (hunk 5 above)
Comment (lathanh) [reply docs/project_defn.rst]: K, I'm just going to remove it from this spot (so documentation isn't out of sync with the state of code) and from the config spec, but I've put todo there.

File: docs/installation.rst (hunk 3 above)
Comment (lathanh) [reply docs/installation.rst:205]: I'm inclined to leave that out... I don't think this is a place the place to hand-hold through every dependency (including Python + venv and Docker, as well as GCP permissions), I also want to somewhat match the level of detail of the rest of the doc and not make it too long, there could be a lot of other factors in how others set up accounts and permissions, and I'd like to minimize the proliferation of TODOs.

But feel free to re-open if you disagree!

**Review by nweires — APPROVED** [review]

**Pull request merged**